conans/model/build_info_components.py

```diff
@@ -0,0 +1,203 @@
+import os
+from collections import OrderedDict
+
+from conans.errors import ConanException
+
+DEFAULT_INCLUDE = "include"
+DEFAULT_LIB = "lib"
+DEFAULT_BIN = "bin"
+DEFAULT_RES = "res"
+DEFAULT_SHARE = "share"
+DEFAULT_BUILD = ""
+
+
+class Component(object):
+
+    def __init__(self, name, root_folder):
+        self._rootpath = root_folder
+        self._name = name
+        self.deps = []
+        self._lib = None
+        self._exe = None
+        self.system_deps = []
+        self.includedirs = [DEFAULT_INCLUDE]
+        self.libdirs = [DEFAULT_LIB]
+        self.resdirs = [DEFAULT_RES]
+        self.bindirs = [DEFAULT_BIN]
+        self.builddirs = [DEFAULT_BUILD]
+        self.srcdirs = []
+        self.defines = []
+        self.cflags = []
+        self.cxxflags = []
+        self.sharedlinkflags = []
+        self.exelinkflags = []
+        self.filter_empty = True
+
+    def as_dict(self):
+        result = {}
+        for name in ["name", "rootpath", "deps", "lib", "exe", "system_deps",
+                     "includedirs", "srcdirs", "libdirs", "bindirs", "builddirs", "resdirs",
+                     "defines", "cflags", "cxxflags", "sharedlinkflags", "exelinkflags"]:
+            result[name] = getattr(self, name)
+        return result
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def rootpath(self):
+        return self._rootpath
+
+    @property
+    def lib(self):
+        return self._lib
+
+    @lib.setter
+    def lib(self, name):
+        assert isinstance(name, str), "'lib' attribute should be a string"
+        if self._exe:
+            raise ConanException("'.exe' is already set for this Component")
+        self._lib = name
+
+    @property
+    def exe(self):
+        return self._exe
+
+    @exe.setter
+    def exe(self, name):
+        assert isinstance(name, str), "'exe' attribute should be a string"
+        if self._lib:
+            raise ConanException("'.lib' is already set for this Component")
+        self._exe = name
+
+
+class DepComponent(object):
+
+    def __init__(self, component):
+        self._rootpath = component.rootpath
```

Comment:
may be it's easier to just store into `self.component`?

Reply:
Yes, it could be done. But I wanted to make the attributes more explicit here